File: local-plugins/fastspell.nvim/jslib/src/types/requests.ts
type CheckSpellRequest = {
    Kind: "check_spell"
    text: string,
    languageId?: string,
    startLine: number
}

type SuggestionRequest = {
    Kind: "suggestion",
    text: string
}

type ConfigureSpellCheckerRequest = {
    Kind: "configure_spell_check_request",
    configFilePath: string
}

type SpellRequest = CheckSpellRequest | SuggestionRequest | ConfigureSpellCheckerRequest;

export type {CheckSpellRequest, SpellRequest, SuggestionRequest, ConfigureSpellCheckerRequest}

File: local-plugins/fastspell.nvim/lua/types/requests.lua
---@class CheckSpellRequest
---@field Kind "check_spell"
---@field text string
---@field languageId string?
---@field startLine number


---@class ConfigureSpellCheckerRequest
---@field Kind "configure_spell_check_request"
---@field configFilePath string


---@alias SpellRequest CheckSpellRequest | ConfigureSpellCheckerRequest
